### ISSUE TITLE
`clone()` stores wrong `url` in `.gitmodules` on windows

### ISSUE THREAD

**Issue by mih** [datalad/datalad#7510]

**Comment by mih** [comment]
Nope, this is not windows. This is a general problem, and replicates in exactly the same fashion on Debian.

The culprit is this line https://github.com/datalad/datalad/blob/f98924576eec8b2629367e466a624b95524e6b02/datalad/support/gitrepo.py#L3808

It takes the URL from the just cloned (sub)dataset -- fine in general. However, if the URL is actually a relative path, this path is relative from the POV of the subdataset. However, it would need to be relative to the superdataset to be correct.

**Comment by adswa** [comment]
It is not the same issue, but I have a feeling that https://github.com/datalad/datalad/issues/7478 is related to this one
